File: 2032-largest-odd-number-in-string/largest-odd-number-in-string.cpp
class Solution {
public:
    string largestOddNumber(string num) {
        
        int n=num.size();
        bool flag=false;
        int ind;

        for(int i=n-1;i>=0;i--){
            int dig=num[i]-'0';
            if(dig%2!=0){
                flag=true;
                ind=i;
                break;
            }
        }

        return flag==false?"":num.substr(0,ind+1);

    }
};

File: 2032-largest-odd-number-in-string/largest-odd-number-in-string.java
class Solution {
    public String largestOddNumber(String num) {
        
        int n=num.length();
        int ind=-1;

        for(int i=n-1;i>=0;i--){
            int dig=num.charAt(i)-'0';
            if(dig%2!=0){
                ind=i;
                break;
            }
        }

        return ind==-1?"":num.substring(0,ind+1);
    }
}